### PR TITLE
Verified GitHub stats

### LINKED ISSUE #10917: Show repository name, in the GitHub statistics section
**What's the problem this feature will solve?**
Currently, it's not possible to discover the name of the GitHub repository, associated with a project's metadata from the PyPI page directly. It is also occasionally not immediately clear which of the URLs associated with a project is a GitHub URL.

This makes it possible for projects to misrepresent what their GitHub statistics are, by using a more popular Python project's URL on their metadata.

**Describe the solution you'd like**
Change "GitHub statistics:" to instead be "GitHub: {repo-name}". So, for pip, that would be "GitHub: pypa/pip".

This would:

- provide an "obvious thing" to make into a link to the GitHub repository.
- surface this information to help identify projects that are misrepresenting themselves (intentionally or otherwise).

**Additional context**
See https://discuss.python.org/t/14181/, where this was originally proposed.

### LINKED ISSUE #14727: Project Details Sidebar Statistics behavior
Eventually replace it with a new statistics sidebar that does the following:

* if trusted publishing from GitHub has been used for the release:
  * include the URL of the upstream repository (https://github.com/pypi/warehouse/issues/10917)
  * fetch statistics for the repository that trusted publishing is configured for
  * mark the statistics as "verified via Trusted Publishing", with a link
* if trusted publishing was not used, and a GitHub URL was provided in the metadata (e.g., we would currently show statistics):
  * don't fetch the statistics
  * instead, include a message + provide a link that indicates that trusted publishing should be used to enable GitHub statistics

_Originally posted by @di in https://github.com/pypi/warehouse/pull/14657#discussion_r1349338677_

### PR DESCRIPTION
**Pull request by di**
This PR moves the "GitHub Statistics" into the "Verified details" section if we have a verified GitHub URL. It does _not_ display the statistics section if there is not a verified URL:

| Before | After |
|--------|--------|
| Screenshot 2024-08-20 at 7 48 13 PM | Screenshot 2024-08-20 at 7 46 35 PM | 

It also adds a link to the repository in the Statistics section and fixes some errant styling in the unverified section.

Fixes #14727, closes #10917.

### CODE REVIEW

**Review by miketheman — COMMENTED** [review]
Generally fine, a couple of comments on tests and style.

File: tests/unit/packaging/test_models.py [hunk 1]
```diff
@@ -819,12 +819,14 @@ def test_acl(self, db_session):
             ("git@bitbucket.org:definex/dsgnutils.git", None),
         ],
     )
-    def test_github_repo_info_url(self, db_session, home_page, expected):
-        release = DBReleaseFactory.create(home_page=home_page)
-        assert release.github_repo_info_url == expected
+    def test_verified_github_repo_info_url(self, db_session, url, expected):
+        release = DBReleaseFactory.create()
+        if url:
+            release.project_urls["Homepage"] = {"url": url, "verified": True}
```
[the remark this answers is not in the document]
Comment (di) [reply tests/unit/packaging/test_models.py:825]: This is testing that `verified_github_repo_info_url` returns `None` when there isn't a verified URL, but I can move that to a separate test.

**Comment by di** [comment]
OK, here's an updated screenshot with the requested changes made:

image

**Review by miketheman — APPROVED** [review]
Let's goooo

**Comment by MartinThoma** [comment]
Thanks for taking care of this :pray: 

Is the verification mechanism described somewhere? Or could you point me to where I can see that in the code :sweat_smile:

**Comment by di** [comment]
See https://docs.pypi.org/project_metadata/

**Pull request merged**